system-addon/content-src/components/PreferencesPane/PreferencesPane.jsx

```diff
@@ -110,10 +110,20 @@ class PreferencesPane extends React.PureComponent {
               {sections
                 .filter(section => !section.shouldHidePref)
                 .map(({id, title, enabled, pref}) =>
-                  <PreferencesInput key={id} className="showSection" prefName={(pref && pref.feed) || id}
-                    value={enabled} onChange={(pref && pref.feed) ? this.handlePrefChange : this.handleSectionChange}
-                    titleString={(pref && pref.titleString) || title} descString={pref && pref.descString} />)}
+                  <div key={id}>
+                    <PreferencesInput className="showSection" prefName={(pref && pref.feed) || id}
+                      value={enabled} onChange={(pref && pref.feed) ? this.handlePrefChange : this.handleSectionChange}
+                      titleString={(pref && pref.titleString) || title} descString={pref && pref.descString} />
 
+                        {pref.nestedPrefs && pref.nestedPrefs.map(nestedPref =>
```

Comment:
This indentation makes it look like these nested prefs are children of `PreferencesInput` and indeed the `options` for TopSites does this same sibling behavior. @rlr noted there was some oddness of not having the nested things as actually nested… I think this was in the context of when we were making the pane scrollable. Not sure if we should just insert `props.children` for `PreferencesInput` ?

Reply:
Good question. We'd need to pass activity stream prefs (or at least the required values) along as well and should probably also move the handler functions. Might be easier to refactor when moving TopSites to a Section to get rid of the special case, but not sure either. I guess it depends on what problem the current nesting is causing?

Reply:
I don't think there's a current bug, but we did run into one during https://github.com/mozilla/activity-stream/pull/3330#discussion_r136465642 where I had set a `section:last-child {margin-bottom}` to make sure the last item is scrollable.. but turns out the `div.options` was actually the last item! Also, there's other pref sections now and we added the padding to the parent instead of margin on the last item, so again, no bug right now…

Reply:
OK, I am up for trying the refactoring. I am thinking in a separate commit/pr though...

Reply:
I think it's fine to have the nestedPrefs be on the same level as the main one for now, but I agree the indentation is a little hard to read/misleading; perhaps you could change it to something like [this](https://gist.github.com/k88hudson/b4684bbce7d2fb181ae1c51b3672bed1)?